packages/terminal/src/widget.ts

```diff
@@ -157,6 +163,42 @@ export class Terminal extends Widget {
     }
   }
 
+  /**
+   * Handle the setting of, connecting to, and sending of (if configured)
+   * initial command on a non-null server session
+   *
+   * @param session - the server session to bind to the terminal
+   * @retuns whether successfully connected to server session
+   *
+   * #### Notes
+   * Subclasses may reimplement this method as needed, but must
+   * `await super.setSession(session)` for proper initialization
+   */
+  protected async setSession(
```

Comment:
Is there a use-case you have in mind here? It doesn't seemed to be used anywhere in lab, and is mostly redundant with the synchronous version above.